File: Oauth2Simples/src/main/java/br/com/estudos/oauth2/repository/BalanceRepository.java
package br.com.estudos.oauth2.repository;

import br.com.estudos.oauth2.model.Balance;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import br.com.estudos.oauth2.dto.ResumeBalance;

import java.util.List;
import java.math.BigDecimal;
import java.util.UUID;


@Repository
public interface BalanceRepository extends CrudRepository<Balance,Long>{
     
     @Query("SELECT new br.com.estudos.oauth2.dto.ResumeBalance(e.typeBalance,sum(e.value)) FROM Balance e " +
            "where e.typeBalance.situation = 'S' AND e.typeBalance.typeCompany = 'BANK' " + 
            "group by e.typeBalance.id")
     List<ResumeBalance> sumBalanceByTypeBalance();

     @Query("SELECT SUM(e.value) from Balance e where e.typeBalance.id = :id")
     BigDecimal findAvailableBalanceByTypeBalance(@Param("id") UUID id);


}

File: kotlin/kotlinEstudos/src/main/java/br/com/kotlin/estudos/repository/BalanceRepository.kt
package br.com.kotlin.estudos.repository

import br.com.kotlin.estudos.model.repository.financial.Balance
import br.com.kotlin.estudos.model.repository.financial.Stock
import br.com.kotlin.estudos.model.repository.financial.Usuario
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface BalanceRepository: CrudRepository<Balance,Long> {

    fun save(balance: Balance)
    fun findByUser(user: Usuario?): List<Balance>?
}